phantom.py

```diff
@@ -0,0 +1,207 @@
+        self.phantomout = phantomout
+        self.scat_density = scat_density
+        self.phantom_bounds = phantom_bounds
+        self.timestep = timestep
+        self.rng_seed = rng_seed
+        self.delta_xyz = delta_xyz
+        self.nodout = nodout
+
+        self.nodeIDcoords = None
+        self.dispdat_header = None
+        self.minmaxs = None
+        self.lims = None
+        self.n_scats = None
+        self.scatterers = None
+        self.translate_scatterers = None
+        self.dispdat = None
+        self.sdispdat = None
+
+        self.load_nodeIDcoords()
+        self.read_dispdat_header()
+        self.calc_n_scats()
+        self.create_scatterers()
+        self.rigid_translate_scatterers()
+        self.make_dispdata()
+        self.make_interp()
+
+
+    def load_nodeIDcoords(self):
+        """read in node IDs and x, y, z coordinates from nodes.dyn mesh file
+
+        """
+        try:
+            from fem.mesh.fem_mesh import load_nodeIDs_coords
+        except ImportError:
+            print('ERROR: Problem importing fem.mesh.fem_mesh.load_nodeIDs_coords; check PYTHONPATH.')
+
+        self.nodeIDcoords = load_nodeIDs_coords(self.nodesdynfile)
+
+    def read_dispdat_header(self):
+        """read in header from disp.dat.xz (or whatever disp.dat binary file)
+
+        header:
+            num_nodes
+            num_dims
+            num_timesteps
+        """
+
+        try:
+            from fem.post.create_res_sim_mat import read_header
+        except ImportError:
+            print('ERROR: Problem importing fem.post.create_res_sim_mat.read_header; check PYTHONPATH.')
+
+        self.dispdat_header = read_header(self.dispdatfile)
+
+    def calc_n_scats(self):
+        """" N (int) - number of scatterers in the model volume
+
+        """
+        i = 0
+        minmaxs = [min(self.nodeIDcoords['x']), max(self.nodeIDcoords['x']),
+                   min(self.nodeIDcoords['y']), max(self.nodeIDcoords['y']),
+                   min(self.nodeIDcoords['z']), max(self.nodeIDcoords['z'])]
+        lims = list()
+
+        # Deal with None cases
+        for item in self.phantom_bounds:
+            for num in item:
+                if num is None:
+                    lims.append(minmaxs[i])
+                else:
+                    lims.append(num)
+                i = i + 1
+
+        tracking_volume = np.abs(lims[1] - lims[0]) *\
+            np.abs(lims[3] - lims[2]) *\
+            np.abs(lims[5] - lims[4])
+
+        self.minmaxs = minmaxs
+        self.lims = lims
+        self.n_scats = int(np.round(tracking_volume*self.scat_density , decimals = 6))
+
+    def create_scatterers(self):
+        """create scatterrers within phantom_bounds using explicitly-seeded RNG
+        Using the explicit RNG seed ensures identical scatterer location in subsequent runs if needed.
+
+        """
+        np.random.seed(self.rng_seed)
+        scatterers = np.random.random((self.n_scats, 3))
+        lims = self.lims
```

Comment:
[nit] any reason to assign to local `lims` variable instead of just using `self.lims` everywhere?

Reply:
no reason, changed this